### ISSUE TITLE
Imported SVG doesn't have correct stroke width

### ISSUE THREAD

**Issue by fsih** [paperjs/paper.js#1514]
# Description/Steps to reproduce
An SVG that contains a path with both a stroke and a scale transform has the stroke width affected by the transform. However when imported to Paper, the stroke width is not affected by the transform. The SVG exported by Paper also no longer transforms the stroke width.

# Link to reproduction test-case
The SVG looks like https://gist.github.com/fsih/71124e912b033a83fca32bf6418daba4

When imported to paper, it looks like http://sketch.paperjs.org/#S/dVNLc5swEP4rO+oBM7XFy02BJM5MD+3FObntoaUHBcugGiRGyIaMx/89K7Bjt001PHb3e2hZ4EAkqzlJyWrLTV6SKcnV2uZ7pqHdF3APmXTuHvq6gj3XrVDyPiMB9TMCXCJXyAIL375+nsUZeViA897yrRIlskWsNKZJPa/rOtpFVOnCC33f95CCHpYOuJyBnfaVkNs3NUGSJN4AX6uuWwoQ6MTalJjFCSYlF0VpLBb5f4gE7z6pHgEffMBm7JmRxYVx1zBTvqZDaSOqChVSSY5erdFqyzF/5w/rtTQ7dxBe7TgYrLH4CGH0AYKPESwhnM+vo/jGRha9RIj+gL99jGay3Shdo1/NjBb9JKHxFPzhGMJZGCZzvAU3ceSi3ju/Fjv0hXObyUw2Wv3muaGibpQ2q+9fJohN4aDkUrF1CpudzA2OdiIMr93D2ICNaaNaYRH8NOwkac6l4RpNj0cXr7mSrYEntZPrFikNa7im590Yeu75kj1jbaScFKritFLF5Ezk/bmtw+XZrxdrV/jsskjB6B2fvk0at0jBwS0Mtun8hzfOMQXJu1PDj+NkXTqMu2KGT2ajG+1xsqfw2f3X7+i6t/gbPWnOto0S0rQk/fnr+AI=

# Expected result
The imported path with a stroke width and a transform should display with a thicker stroke width in Paper. Perhaps the path should be treated as applyMatrix = false, or when applying the matrix the stroke width should be scaled.

# Additional information

**Comment by sasensi** [comment]
Hi,

you can control wether matrix is applied or not by switching global `applyMatrix` setting before importing items.
Here is the modified [Sketch](http://sketch.paperjs.org/#S/dVNNc5swEP0rGnoAprbA2EkNSZyZHtqLc3KbQ0sPCl6DapAYSTZ4PP7vWYGJ3TbV8LG77+3bZYWOjmAVOImz2oLJCmfkZHJtfQ3GcJFryuq6PDwxo3hLHsiGlRruUpGKPVNE73OMpcK9f2yrkuxBaS7FQ+pMaJg6BASKoQgGvn/7Mp6nzuOCuB8t32ZiitCIFcbUSRA0TUObKZUqD6IwDAOkoIalE1xux07akovtuzmTOI6DDr7Oum5pgkDD16ZAbx6jUwDPC2OxafhHEofms2wRCElIsBl7p87iwrivmSne3C604WWJGUIKQC1tlNwC+h/Cbr2FxkMH0VXFTmCNwScSTW/I5NOULEk0m11b81trWfRiIfqD/K1jFBN6I1WFelW3b15M5yMSdldnjqMonuFrcjuf+pgfDNtih75wu/2tlfwNmaG8qqUyq+evHmIjcpRiKdk6IZudyAyO1uMGKv/YN2BtWkvNLYK/hp0kzUAYUCh6Ovn4zKTQhrzInVhrpNSsBkWHagw197BkB4z1lHOGLIGWMvcGIrRDW8fLt18vplf47SJPiFE7GL1P6kskxMUSBtt0/8Pr55gQAc254f5EeD7txl0yA964V6MtTvZsHvx/9U6+f4fn7EUB29aSC6Od5Oev0ys=) where stroke is properly scaled.

The problem you encountered is linked to the fact that, at the end of the import process, `applyMatrix` is set, for imported items, to the value it was when import started.
https://github.com/paperjs/paper.js/blob/6cbf5292da1551cd870a3ef78e087199d2344738/src/svg/SvgImport.js#L640-L643

**Comment by fsih** [comment]
@sasensi but doesn't this indicate that there is a problem with applyMatrix? Applying the matrix shouldn't significantly change the appearance. Globally not applying the matrix affects all other shapes in the SVG which don't require that change.

**Comment by sasensi** [comment]
I agree with you about the fact that applying the matrix or not shouldn't change the appearance of the imported items.

Unfortunately, I can't see a simple solution for this:

- forcing `applyMatrix` to `false` for SVG nodes that have a transform attribute may lead to incoherent behavior, having some imported items with matrix applied and other without.
- applying scale transformation to stroke width is also problematic because if horizontal scale amount is different from vertical one, we can't just rely on `strokeWidth` property, but need to store that transformation in something like a independant stroke matrix instead.

Does someone have an idea ?

**Comment by fsih** [comment]
applyMatrix already does not propagate to Raster, PointText, and SymbolItem at the leaf level, by having _transformContent return false: https://github.com/paperjs/paper.js/blob/a61e83edf2ed1870bd41bad135f4f6fc85b0f628/src/item/Item.js#L3464
but I agree that it could be confusing to have paths inconsistently have the transform stored in their matrix. Scaling the strokes by an average of the x and y scales might be incrementally better. Scaling x and y by the same amount is probably the most common case.
